app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,57 @@
+class ArticlesController < ApplicationController
+  before_action :find_article, except: %i[index new create]
+  http_basic_authenticate_with name: 'user', password: 'secret', except: %i[index show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show; end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+    begin
+      @article.save!
+    rescue ActiveRecord::RecordInvalid
+      puts e.message
```

Comment:
this will raise an error because `e` does not exist.

Reply:
My bad, now it's deleted